src/auro-input.js

```diff
@@ -3,55 +3,161 @@
+import { html } from "lit-element";
+import { classMap } from 'lit-html/directives/class-map.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import BaseInput from './base-input';
 
-// If using auroElement base class
-// See instructions for importing auroElement base class https://git.io/JULq4
-// import { html, css } from "lit-element";
-// import AuroElement from '@alaskaairux/orion-web-core-style-sheets/dist/auroElement/auroElement';
 
-// Import touch detection lib
-import "focus-visible/dist/focus-visible.min.js";
-import styleCss from "./style-css.js";
+// build the component class
+export default class AuroInput extends BaseInput {
 
-// See https://git.io/JJ6SJ for "How to document your components using JSDoc"
-/**
- * auro-input provides users a way to ...
- *
- * @attr {String} cssClass - Applies designated CSS class to DOM element.
- */
+  constructor() {
+    super();
 
-// build the component class
-class AuroInput extends LitElement {
-  // constructor() {
-  //   super();
-  // }
+    /**
+     * @private Boolean value to determine which password icon to show
+     */
+    this.showPassword = false;
+  }
 
   // function to define props used within the scope of this component
   static get properties() {
     return {
-      // ...super.properties,
-      cssClass:   { type: String }
+      ...super.properties,
     };
   }
 
-  static get styles() {
-    return css`
-      ${styleCss}
-    `;
+  /**
+   * @private custom setter so we can re-validate on update
+   * @param {string} value Error string
+   */
+  set error(value) {
+
+    const oldVal = this._error;
+
+    this._error = value;
+    this.requestUpdate('error', oldVal).then(this.validate.bind(this));
   }
 
-  // When using auroElement, use the following attribute and function when hiding content from screen readers.
-  // aria-hidden="${this.hideAudible(this.hiddenAudible)}"
+  get error() {
+    return this._error;
+  }
+
+  /**
+   * @private function to toggle between text or password view of input
+   * @returns {string} returns string based on type
+   */
+  handleClickShowPassword() {
+    const el = this.shadowRoot.getElementById(`${this.id}`);
+
+    if (this.type === "password") {
+      this.type = "text";
+      this.showPassword = true;
+      el.focus();
+    } else {
+      this.type = "password";
+      this.showPassword = false;
+      el.focus();
+    }
+  }
+
+  /**
+   * @private function to toggle between password icons
+   * @returns {string} returns HTML for SVG
+   */
+  togglePasswordIcon() {
+    if (this.showPassword) {
+      return this.hidePassword
+    }
+
+    return this.viewPassword
+  }
+
+  /**
+   * @private function manage visibility of show-password icon
+   * @returns {boolean} returns CSS selector
+   */
+  handleKeyUp() {
+    const inputValue = this.shadowRoot.getElementById(this.id).value
+
+    if (inputValue) {
+      this.classList.add("passwordIcon--show");
```

Comment:
It seems weird to expose classes like this on the host <auro-input>, where the consumer can see/interact with. Is there a way to move this within the shadow root?

Reply:
refactored